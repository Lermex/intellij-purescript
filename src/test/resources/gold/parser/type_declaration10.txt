Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Wii')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('String')
                          PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('String')
                  PsiElement())(')')
                  PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeVar)
                PSASTWrapperElement(TypeConstructor)
                  <empty list>
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('eff')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Wii')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('String')
                          PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('String')
                  PsiElement())(')')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Wii')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('Array')
                          PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PsiElement(()('(')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('User')
                              PsiElement(whitespace)(' ')
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(GenericIdentifier)
                            PsiElement(identifier)('a')
                      PsiElement())(')')
                  PsiElement())(')')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Wii')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(GenericIdentifier)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(GenericIdentifier)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)(' ')
                  PsiElement(->)('->')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('c')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('d')
                          PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('User')
                              PsiElement(whitespace)(' ')
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(GenericIdentifier)
                            PsiElement(identifier)('a')
                            PsiElement(whitespace)(' ')
                        PsiElement(->)('->')
                        PsiElement(whitespace)(' ')
                        PSASTWrapperElement(Type)
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(TypeConstructor)
                              PSASTWrapperElement(Qualified)
                                PsiElement(proper name)('B')
            PsiElement())(')')