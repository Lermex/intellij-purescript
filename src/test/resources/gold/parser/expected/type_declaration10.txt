Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Wii')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('console')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('String')
                          PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('String')
                  PsiElement())(')')
                  PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('eff')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Wii')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('console')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('String')
                          PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('String')
                  PsiElement())(')')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Wii')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('console')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('Array')
                          PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PsiElement(()('(')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('User')
                              PsiElement(whitespace)(' ')
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeVar)
                            PsiElement(identifier)('a')
                      PsiElement())(')')
                  PsiElement())(')')
            PsiElement())(')')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Wii')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('console')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeVar)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeVar)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)(' ')
                  PsiElement(->)('->')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('c')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('d')
                          PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('User')
                              PsiElement(whitespace)(' ')
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeVar)
                            PsiElement(identifier)('a')
                            PsiElement(whitespace)(' ')
                        PsiElement(->)('->')
                        PsiElement(whitespace)(' ')
                        PSASTWrapperElement(Type)
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(TypeConstructor)
                              PSASTWrapperElement(Qualified)
                                PsiElement(proper name)('B')
            PsiElement())(')')